frontend/apps/reader/readerui.lua

```diff
@@ -593,28 +593,66 @@ end
 function ReaderUI:showReader(file, provider, seamless)
     logger.dbg("show reader ui")
 
-    file = ffiUtil.realpath(file)
     if lfs.attributes(file, "mode") ~= "file" then
         UIManager:show(InfoMessage:new{
              text = T(_("File '%1' does not exist."), BD.filepath(filemanagerutil.abbreviate(file)))
         })
         return
     end
 
-    if not DocumentRegistry:hasProvider(file) and provider == nil then
+    if provider == nil and DocumentRegistry:hasProvider(file) then
+        provider = DocumentRegistry:getProvider(file)
+    end
+    if provider ~= nil then
+        provider = self:extendProvider(file, provider)
+    end
+    if provider and provider.provider then
+        -- We can now signal the existing ReaderUI/FileManager instances that it's time to go bye-bye...
+        UIManager:broadcastEvent(Event:new("ShowingReader"))
+        self:showReaderCoroutine(file, provider, seamless)
+    else
         UIManager:show(InfoMessage:new{
             text = T(_("File '%1' is not supported."), BD.filepath(filemanagerutil.abbreviate(file)))
         })
         self:showFileManager(file)
-        return
     end
+end
 
-    -- We can now signal the existing ReaderUI/FileManager instances that it's time to go bye-bye...
-    UIManager:broadcastEvent(Event:new("ShowingReader"))
-    provider = provider or DocumentRegistry:getProvider(file)
-    if provider.provider then
-        self:showReaderCoroutine(file, provider, seamless)
+function ReaderUI:extendProvider(file, provider)
+    -- If file extension is pure "zip", check the archive content and choose the appropriate provider,
+    -- except when the provider choice is forced in the "Open with" dialog.
+    -- Also pass to crengine is_fb2 property, based on the archive content (pure "zip"),
```

Comment:
`pure "zip"` sounds odd to read. `only "zip"`, `solely "zip"` ?

Reply:
I'm not sure if there's a word for them but something like single/simple vs double maybe?